loader/src/reference_binder.py

```diff
@@ -0,0 +1,150 @@
+import re
+from collections import defaultdict
+from dotty_dict import dotty
+
+from loader.src.config.logger import create_logger
+
+logger = create_logger("reference_binder")
+
+
+# dotty-dict does not handle brackets indices,
+# it uses dots instead (a.0.b instead of a[0].b)
+def dotty_paths(paths):
+    for path in paths:
+        yield re.sub(r"\[(\d+)\]", r".\1", path)
+
+
+class ReferenceBinder:
+    def __init__(self, fhirstore):
+        self.fhirstore = fhirstore
+
+        # cache is a dict of form
```

Comment:
I think you forgot to add the type values

Reply:
done